### PR TITLE
Extend contributing with add/deprecate examples guidelines

### PR DESCRIPTION
**Pull request by maxproske**
Extended the example contribution guidelines for consistency, and learnings I've made as a new contributor.

I also moved general guidelines out of the `README.md`-specific section.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -265,26 +265,27 @@ Below are the steps to add a new link:
 
 When you add an example to the [examples](examples) directory, please follow these guidelines to ensure high quality examples:
 
-- TypeScript should be leveraged for new examples (no need for separate JavaScript and TypeScript examples, converting old JavaScript examples is preferred)
-- Examples should not add custom ESLint configuration (we have specific templates for ESLint)
+- TypeScript should be leveraged for new examples (no need for separate JavaScript and TypeScript examples, converting old JavaScript examples is preferred as it's less to maintain). TypeScript helps eliminate a class of bugs while also providing better developer experience, even for newcomers.
+- Examples should not add custom ESLint or Prettier configurations (we have specific templates for those)
```
Comment (maxproske) [inline contributing.md:269]: x-ref: https://github.com/vercel/next.js/pull/38504#discussion_r917839211

File: contributing.md [hunk 2]
```diff
@@ -265,26 +265,27 @@ Below are the steps to add a new link:
...
 - If API routes aren't used in an example, they should be omitted
 - If an example exists for a certain library and you would like to showcase a specific feature of that library, the existing example should be updated (instead of adding a new example)
 - Package manager specific config should not be added (e.g. `resolutions` in `package.json`)
 - In `package.json` the version of `next` (and `eslint-config-next`) should be `latest`
-- In `package.json` the dependency versions should be up-to-date
+- In `package.json` the dependency versions should be up-to-date, and accept new minor and patch versions (e.g. `^version`)
+- Omit the `name` and `version` fields from your `package.json`
 - Use `export default function` for page components and API Routes instead of `const`/`let` (The exception is if the page has `getInitialProps`, in which case [`NextPage`](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props#typescript) could be useful)
+- TypeScript examples should not use `prop-types`
+- TypeScript `strict` mode should be set to `false` in `tsconfig.json`
```
Comment (maxproske) [inline contributing.md:278]: x-ref: https://github.com/vercel/next.js/pull/37676#issuecomment-1154894519

File: contributing.md [hunk 3]
```diff
@@ -265,26 +265,27 @@ Below are the steps to add a new link:
...
+- Examples should not add custom ESLint or Prettier configurations (we have specific templates for those)
 - If API routes aren't used in an example, they should be omitted
 - If an example exists for a certain library and you would like to showcase a specific feature of that library, the existing example should be updated (instead of adding a new example)
 - Package manager specific config should not be added (e.g. `resolutions` in `package.json`)
 - In `package.json` the version of `next` (and `eslint-config-next`) should be `latest`
-- In `package.json` the dependency versions should be up-to-date
+- In `package.json` the dependency versions should be up-to-date, and accept new minor and patch versions (e.g. `^version`)
+- Omit the `name` and `version` fields from your `package.json`
 - Use `export default function` for page components and API Routes instead of `const`/`let` (The exception is if the page has `getInitialProps`, in which case [`NextPage`](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props#typescript) could be useful)
+- TypeScript examples should not use `prop-types`
```
Comment (maxproske) [inline contributing.md:277]: x-ref: https://github.com/vercel/next.js/pull/38021#issuecomment-1166491276

File: contributing.md [hunk 4]
```diff
@@ -265,26 +265,27 @@ Below are the steps to add a new link:
...
+- TypeScript `strict` mode should be set to `false` in `tsconfig.json`
+- Type definition files should be inside the `interfaces` directory. `.d.ts` modules should be in the root directory
```
Comment (maxproske) [inline contributing.md:279]: This is a tricky one. The Next.js and React docs don't appear to recommend a location for type definition files. Should the Contribution docs reccomend a directory for consistency? There are currently 13 examples that include type definition files:

- `interfaces/*.ts` 6 examples (including `with-typescript`, mostly from me copying this pattern)
- `types/*.ts`: 1 example (including `with-typescript-types`, and my preference)
- `*.d.ts`: 3 examples (e.g. `environment.d.ts` for environment variables and other globals)
- `components/types.ts`: 1 example (co-located with components?)
- `lib/types.ts`: 1 example 
- `@types/*.ts`: 1 example

File: contributing.md [hunk 5]
```diff
@@ -315,6 +316,16 @@ pnpm create next-app --example DIRECTORY_NAME DIRECTORY_NAME-app
...
+## Deprecating examples
+
+Replace the app with just a `README.md` file.
+
+```markdown
+## Deprecated
+
+The main [blog-starter](/examples/blog-starter) example has been refactored to use TypeScript, so this example is deprecated.
+```
+
```
Comment (maxproske) [inline contributing.md:328]: x-ref: https://github.com/vercel/next.js/pull/38395

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks for clarifying some of these! I added some comments. In general, we don't want to push our opinions too much when unnecessary, but most of your suggestions make sense! :+1:

File: contributing.md (hunk 4 above)
Comment (maxproske) [reply contributing.md]: Makes sense! I'll remove it for now.

**Pull request closed without merging**